superset/assets/src/explore/components/ExploreViewContainer.jsx

```diff
@@ -106,7 +106,6 @@ class ExploreViewContainer extends React.Component {
   componentWillReceiveProps(nextProps) {
     if (nextProps.controls.viz_type.value !== this.props.controls.viz_type.value) {
       this.props.actions.resetControls();
-      this.props.actions.triggerQuery(true, this.props.chart.id);
```

Comment:
Sorry for the delay, this change is to fix a bug in the master branch (0.32 release).
when user change viz_type (or any control that needs trigger new query), we should show a overlay to hold the change, and allow user to update more controls change. when user clicks on `Run query` we trigger new query with all change together. But in current master, change viz_type will trigger new query right away (but not update UI because of overlay): 

![B48ewQojQN](https://user-images.githubusercontent.com/27990562/57156844-dacd0480-6d93-11e9-953a-c5aa488dfdec.gif)

This line of change (not setting `triggerQuery`) is just to fix this additional query issue.